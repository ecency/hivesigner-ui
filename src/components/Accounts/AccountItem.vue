<template>
  <div class="flex items-center p-3 border rounded overflow-hidden mb-3 justify-between">
    <div class="flex w-full items-center">
      <h4 class="flex items-center text-black-500 text-xl font-bold my-1.5">
        <Avatar :username="user" class="mr-2"/>
        {{ user }}
      </h4>
      <span v-if="user === username">
        <span class="ml-3 text-black-400">Unlocked</span>
        <router-link class="ml-3 hover:underline cursor-pointer" to="/auths">Auths</router-link>
        <a
          class="ml-3 text-primary hover:text-primary-dark cursor-pointer hover:underline"
          @click="logout"
        >
          Log out
        </a>
      </span>
      <span v-else>
        <router-link class="ml-3 hover:underline cursor-pointer"
                     to="/login?redirect=accounts">
          Unlock
        </router-link>
      </span>
    </div>
    <a
      class="float-right text-primary hover:text-primary-dark p-2 cursor-pointer"
      @click="removeAccount(user)"
    >
      <Icon name="Trash"/>
    </a>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator'
import { AuthModule } from '~/store'
import { removeFromKeychain } from '~/utils'
import Icon from '~/components/UI/Icons/Icon.vue'
import Avatar from '~/components/Avatar.vue'

@Component({
  components: { Avatar, Icon }
})
export default class AccountItem extends Vue {
  @Prop({
    type: String,
    default: '',
  })
  private user!: string

  private get username(): string {
    return AuthModule.username
  }

  private logout(): Promise<void> {
    return AuthModule.logout()
  }

  private removeAccount(username): void {
    removeFromKeychain(username)
    this.$emit('load-keychain')
  }
}
</script>
