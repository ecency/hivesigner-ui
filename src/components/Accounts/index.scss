.AccountsContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 10vh;
}
.Header {
  display: flex;
  flex-direction: column;
  margin-top: 10vh;
  align-items: center;
  span {
    margin-top: 10px;
    font-size: 2.5rem;
    color: red;
  }
}
.AddButton {
  border: 1px solid grey;
  border-radius: 10px;
  padding: 15px 100px;
  max-width: 500px;
  opacity: 0.8;
  margin-top: 25vh;
  cursor: pointer;
  margin-bottom: 7rem;
}
.AccountMenubar {
  font-size: 2rem;
  color: black;
  position: absolute;
  right: 30px;
  top: 30px;
  display: block;
  @media screen and (min-width: 630px) {
    display: none;
  }
}
.Accountfix {
  * {
    top: 30px !important;
  }
}
