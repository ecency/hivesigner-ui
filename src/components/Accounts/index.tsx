import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import { ImMenu } from "react-icons/im";
import { AiOutlineClose } from "react-icons/ai";
import "./index.scss";
import Content from "../../layouts/content";
import Navbar from "../Index/Navbar";
import HomeLogo from "../../assets/img/logo";

const Index = () => {
  const [showNavbar, setshowNavbar] = useState(true);
  const handleScreenChange = (mediaQuery: any) => {
    if (mediaQuery.matches) {
      setshowNavbar(true);
    } else {
      setshowNavbar(false);
    }
  };
  useEffect(() => {
    let window_width = window.matchMedia("(min-width:632px)");
    window_width.addListener(handleScreenChange);
    handleScreenChange(window_width);
    return () => {
      window_width.removeListener(handleScreenChange);
    };
  }, []);

  const handleChange = () => {
    setshowNavbar(!showNavbar);
  };
  return (
    <Content>
      <div className="AccountsContainer">
        <div className="AccountMenubar" onClick={() => handleChange()}>
          {showNavbar ? <AiOutlineClose /> : <ImMenu />}
        </div>
        <div className="Header">
          <div className="account_logo">
            <HomeLogo />
          </div>
          <span>Accounts</span>
        </div>
        <div className="AddButton">
          <Link to="/import">Add another account</Link>
        </div>
        {showNavbar && (
          <div className="Accountfix">
            <Navbar />
          </div>
        )}
      </div>
    </Content>
  );
};

export default Index;
