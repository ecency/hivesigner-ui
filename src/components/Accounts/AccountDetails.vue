<template>
<div class="account-details h-full flex flex-col justify-center">
  <account-item :user="account" />
  <div class="accounts-details-menu flex flex-col text-center text-black-400 mt-12">
    <router-link to="/auths" class="p-4 hover:bg-gray-100">
      {{ $t('accounts.auths') }}
    </router-link>
    <router-link to="/sign/transactions" class="p-4 hover:bg-gray-100">
      {{ $t('accounts.sign_transactions') }}
    </router-link>
    <a
      role="button"
      class="p-4 hover:bg-gray-100"
      @click="confirmModalRef.show()"
    >
      {{ $t('accounts.delete') }}
    </a>
  </div>
  <confirm-modal
    ref="confirm-modal"
    message="accounts.delete_account_confirm"
    positive-label="accounts.delete_account"
    @positive="removeAccount"
  />
</div>
</template>

<script lang="ts">
import { Component, Emit, Prop, Vue, Ref } from 'nuxt-property-decorator'
import AccountItem from './AccountItem.vue'
import { AccountsModule, AuthModule } from '~/store'
import ConfirmModal from '../UI/ConfirmModal.vue'

@Component({
  components: { ConfirmModal, AccountItem },
})
export default class AccountDetails extends Vue {
  @Prop({
    type: String,
    required: true,
  })
  private account!: string

  @Ref('confirm-modal')
  private confirmModalRef!: ConfirmModal

  private async mounted(): Promise<void> {
    try {
      await AuthModule.login({
        username: this.account,
        keys: await AccountsModule.getEncryptedKeys({ username: this.account })
      })
      await AccountsModule.setSelectedAccount(this.account)
    } catch (_) {
    }
  }

  @Emit('removed')
  private removeAccount(): void {
    AccountsModule.removeAccount(this.account)
  }
}
</script>

<style scoped>

</style>
