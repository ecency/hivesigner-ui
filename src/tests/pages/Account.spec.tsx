import React from "react";
import Accounts from "../../components/Accounts";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
describe("Accounts component Test", () => {
  Enzyme.configure({ adapter: new Adapter() });
  test("Header snapshot testing", () => {
    const header = renderer.create(<div className="Header" />).toJSON();
    expect(header).toMatchSnapshot();
  });
  test("menubar snapshot testing", () => {
    const menubar = renderer
      .create(<div className="AccountMenubar" />)
      .toJSON();
    expect(menubar).toMatchSnapshot();
  });

  test("each components test", () => {
    expect(shallow(<Accounts />).find(".Accountfix").length).toEqual(1);
  });
  test("each components test", () => {
    expect(shallow(<Accounts />).find(".AddButton").length).toEqual(1);
  });
  test("each components test", () => {
    expect(shallow(<Accounts />).find(".account_logo").length).toEqual(1);
  });
  test("each components test", () => {
    expect(shallow(<Accounts />).find(".AccountMenubar").length).toEqual(1);
  });
  test("each components test", () => {
    expect(shallow(<Accounts />).find(".AccountsContainer").length).toEqual(1);
  });
});
