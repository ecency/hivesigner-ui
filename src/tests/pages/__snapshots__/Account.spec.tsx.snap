// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accounts component Test Header snapshot testing 1`] = `
<div
  className="Header"
/>
`;

exports[`Accounts component Test menubar snapshot testing 1`] = `
<div
  className="AccountMenubar"
/>
`;
